### ISSUE TITLE
Add a failure handler for serde

### ISSUE THREAD

**Issue by iravid** [zio/zio-kafka#50]
In @svroonland's words:
```
I can see three desirable ways of handling deserialization failures:

    Fail the stream
    Skip the message
    Allow the client to handle it explicitly

1 should be easy (already the case?)
3 can be done by sending Eithers like you mentioned.
Maybe 2 can be implemented with some sort of onDeserializationError: DeserializationFailure => UIO[Unit] parameter on the plainStream and partitionedStream methods?
```

**Comment by svroonland** [comment]
Was thinking about this one a bit more. I personally don't like a sort of "registerExceptionHandler" or adding a parameter `onDeserializationError` to `plainStream`. Would it be sufficient to do the following:

* Fail the stream on deserialization errors by default (current behavior)
* Clients that want to skip non-deserializable messages can use the existing `Deserializer.asTry` in combination with a `mapConcat` if desired. We can write some documentation on how to achieve this behavior.

**Comment by svroonland** [comment]
Implemented in #89
